packages/flutter_tools/lib/src/macos/macos_device.dart

```diff
@@ -59,22 +62,38 @@ class MacOSDevice extends Device {
     bool applicationNeedsRebuild = false,
     bool usesTerminalUi = true,
     bool ipv6 = false,
-  }) {
-    throw UnimplementedError();
+  }) async {
+    if (!prebuiltApplication) {
+      return LaunchResult.failed();
+    }
+    // The location of these files is based on the examples in the flutter
+    // desktop embedding repo and may not be final.
```

Comment:
The `Contents/MacOS` part is intrinsic to macOS apps, not a function of FDE.

It's also standard for the executable for `Foo.app` to be called `Foo`, but end users can rename the `.app` part at any time. The robust way to construct the path would be to get the `CFBundleExecutable` value from `Contents/Info.plist` (that path is also defined by macOS application bundle structure) and use that as the final path component.

Reply:
Done